orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+        selected_rows = [
+            row_index for row_index, (doc_index, _) in enumerate(word_index)
+            if doc_index in selected_docs]
+        selection = QItemSelection()
+        # select all rows belonging to the selected document
+        for row in selected_rows:
+            index = self.model().index(row, 0)
+            selection.select(index, index)
+        super().select(selection, flags)
+
+
+class ConcordanceModel(QAbstractTableModel):
+    "A model for constructing concordances from text."
+
+    def __init__(self):
+        QAbstractTableModel.__init__(self)
+        self.word = None
+        self.data = None
+        self.indices = None
+        self.word_index = None
+        self.width = 8
+
+    def set_word(self, word):
+        self.modelAboutToBeReset.emit()
+        self.word = word
+        if self.indices is None:
+            self.word_index = None
+        else:
+            self.word_index = [
+                (doc_idx, offset) for doc_idx, doc in enumerate(self.indices)
+                 for offset in doc.offsets(self.word)]
+        self.modelReset.emit()
+
+    def set_data(self, data):
+        self.modelAboutToBeReset.emit()
+        self.data = data
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def set_width(self, width):
+        self.modelAboutToBeReset.emit()
+        self.width = width
+        self.modelReset.emit()
+
+    def flags(self, _):
+        return Qt.ItemIsEnabled | Qt.ItemIsSelectable
+
+    def rowCount(self, parent=None, *args, **kwargs):
+        return 0 if parent.isValid() or self.word_index is None \
+               else len(self.word_index)
+
+    def columnCount(self, parent=None, *args, **kwargs):
+        return 3
+
+    def data(self, index, role=Qt.DisplayRole):
+        row, col = index.row(), index.column()
+        doc, index = self.word_index[row]
+
+        if role == Qt.DisplayRole:
+            tokens = self.data.tokens
+            if col == 0:
+                # one more if necessary to assure as much text as possible
+                # is returned
+                return ' '.join(tokens[doc][max(index - self.width, 0):index])
+            if col == 1:
+                return tokens[doc][index]
+            if col == 2:
+                return ' '.join(tokens[doc][index + 1:index + self.width])
+
+        elif role == Qt.TextAlignmentRole:
+            return [Qt.AlignRight | Qt.AlignVCenter,
+                    Qt.AlignCenter,
+                    Qt.AlignLeft | Qt.AlignVCenter][col]
+
+        elif role == Qt.ForegroundRole:
+             if col == 1:
+                 return QBrush(QColor(0, 0, 255))
+
+        elif role == Qt.BackgroundRole:
+            row_color = tuple(map(lambda x:
+                                  x + 23 * (self.word_index[row][0] % 2),
+                                            (232, 232, 232)))
+            return QColor(row_color[0], row_color[1], row_color[2])
```

Comment:
You'll change this code anyway, but for the next time: `return QColor(*row_color)` has the same effect. Plus, as much as I like `map`, it's only better than list comprehension if you already have a function that your going to use for mapping. If you need to define a lambda-function, I'd prefer list comprehension, e.g.

    row_color = [x + 23 * (self.word_index[row][0] % 2) for x in (232, 232, 232)]

Doesn't make sense here, anyway. :)